src/sage/categories/pushout.py

```diff
@@ -3249,7 +3249,11 @@ def __init__(self, polys, names, embeddings=None, structures=None,
                 if latex_names[i] == latex_variable_name(name):
                     latex_names[i] = None
         self.latex_names = latex_names
-        self.kwds = kwds
+        kwds_self = dict(kwds.items())
+        if 'implementation' in kwds_self:
+            self.implementations[0] = kwds_self['implementation']
+            del kwds_self['implementation']
+        self.kwds = kwds_self
 
```

Comment:
Now I see it from https://github.com/sagemath/sage/pull/35609#issuecomment-1536439262

Reply:
Instead of updating `self.implementations[0]` which obscures the point, I suggest
```diff
--- a/src/sage/categories/pushout.py
+++ b/src/sage/categories/pushout.py
@@ -3314,7 +3314,8 @@ class AlgebraicExtensionFunctor(ConstructionFunctor):
         self.latex_names = latex_names
         kwds_self = dict(kwds.items())
         if 'implementation' in kwds_self:
-            self.implementations[0] = kwds_self['implementation']
+            assert len(self.polys) == 1
+            self.implementations = [kwds_self['implementation']]
             del kwds_self['implementation']
         self.kwds = kwds_self
```
since the presence of 'implementation' in `kwds`, I think, assumes there is one algebraic extension by one `poly`.

Reply:
Added suggested change